src/connection.rs

```diff
@@ -416,6 +403,33 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Connection<T> {
                     Either::Right(io)
                 };
 
+            match next_io_event.await? {
+                IoEvent::OutboundReady => {
+                    // Only unpause the control command receiver if we're not
+                    // shutting down already.
+                    if let Shutdown::NotStarted = self.shutdown {
+                        self.control_receiver.unpause();
+                    }
```

Comment:
```suggestion
```
Unless I'm missing something, since the control command receiver is no longer paused when we have a pending write (instead we just don't poll it), there is really nothing to do here. Other than that the control receiver is only paused for shutdown, which we don't want to interfere with anyway.

Reply:
Good point :+1: